Game/world/worldobjects.cpp

```diff
@@ -523,22 +523,41 @@ Bullet& WorldObjects::shootBullet(const Item& itmId,
   }
 
 Item *WorldObjects::addItem(size_t itemInstance, const char *at) {
-  if(itemInstance==size_t(-1))
-    return nullptr;
+  Item* item = nullptr;
+  Tempest::Vec3 pos;
+  Tempest::Vec3 dir;
+  const WayPoint* waypoint = owner.findPoint(at);
+
+  if(waypoint != nullptr) {
+    pos = {waypoint->x, waypoint->y, waypoint->z};
+    dir = {waypoint->dirX, waypoint->dirY, waypoint->dirZ};
+  } else {
+    pos = {0,0,0};
```

Comment:
Vec3 is initialized as {0,0,0} by default, no need for else-path

Reply:
OK, good to know, will be removed.